src/main/java/com/icthh/xm/ms/mstemplate/config/client/OAuth2InterceptedFeignConfiguration.java

```diff
@@ -0,0 +1,81 @@
+package com.icthh.xm.ms.mstemplate.config.client;
+
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import feign.RequestInterceptor;
+import lombok.RequiredArgsConstructor;
+import org.springframework.cloud.openfeign.EnableFeignClients;
+import org.springframework.cloud.openfeign.FeignClientsConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.security.oauth2.client.AuthorizedClientServiceOAuth2AuthorizedClientManager;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProvider;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder.ClientCredentialsGrantBuilder;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
+import org.springframework.security.oauth2.client.endpoint.DefaultClientCredentialsTokenResponseClient;
+import org.springframework.security.oauth2.client.endpoint.OAuth2ClientCredentialsGrantRequestEntityConverter;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+
+@Configuration
+@RequiredArgsConstructor
+@EnableFeignClients(basePackages = "com.icthh.xm.ms.mstemplate.client")
+@Import(FeignClientsConfiguration.class)
+public class OAuth2InterceptedFeignConfiguration {
```

Comment:
Maybe better to create commons that add support feign, to avoid copy paste on each ms

Reply:
Should it be a separate commons dependency or add it to the commons security?

Reply:
I think it is easier to have a separate commons